clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -1640,6 +1640,55 @@ mlir::Value buildNeonCall(unsigned int builtinID, CIRGenFunction &cgf,
   }
 }
 
+mlir::Value CIRGenFunction::buildCommonNeonBuiltinExpr(
+    unsigned builtinID, unsigned llvmIntrinsic, unsigned altLLVMIntrinsic,
+    const char *nameHint, unsigned modifier, const CallExpr *e,
+    llvm::SmallVectorImpl<mlir::Value> &ops, cir::Address ptrOp0,
+    cir::Address ptrOp1, llvm::Triple::ArchType arch) {
+  // Get the last argument, which specifies the vector type.
+  const clang::Expr *arg = e->getArg(e->getNumArgs() - 1);
+  std::optional<llvm::APSInt> neonTypeConst =
+      arg->getIntegerConstantExpr(getContext());
+  if (!neonTypeConst)
+    return nullptr;
+
+  // Determine the type of this overloaded NEON intrinsic.
+  NeonTypeFlags neonType(neonTypeConst->getZExtValue());
+  bool isUnsigned = neonType.isUnsigned();
+  bool isQuad = neonType.isQuad();
+  const bool hasLegalHalfType = getTarget().hasLegalHalfType();
+  assert(!MissingFeatures::allowBFloatArgsAndRet());
+  // The value of allowBFloatArgsAndRet is true for AArch64, but it should
+  // come from ABI info.
+  const bool allowBFloatArgsAndRet = true;
+
+  mlir::Type vTy = GetNeonType(this, neonType, hasLegalHalfType, false,
+                               allowBFloatArgsAndRet);
+  if (!vTy)
+    return nullptr;
+
+  unsigned intrinicId = llvmIntrinsic;
+  if ((modifier & UnsignedAlts) && !isUnsigned)
+    intrinicId = altLLVMIntrinsic;
+
+  switch (builtinID) {
+  default:
+    llvm_unreachable("NYI");
```

Comment:
Not for this PR, but I'd like to see alternative ways for us to early track what's missing, if this assert is the only place, we still have to find what's the name of missing intrinsic with a testcase, which is rather silly and time consuming.

I suggest two different options:
1. Get inspired by `GET_NEON_OVERLOAD_CHECK` and change tablegen to emit a switch that writes the intrinsic name as a string so we can print it before we call unrecheable.
2. Add the skeleton for all missing intrinsics in OG for buildCommonNeonBuiltinExpr, and for each `case`, instead of just `llvm_unreachable("NYI");`, we should have `llvm_unreachable("<name of the intrinsic> NYI");`.

Reply:
that's exactly I was thinking about as it's painful not knowing what the missing intrinsic name is. 
I'll investigate how to do it with option 1